src/plugins/navigation_embeddable/public/components/editor/navigation_embeddable_link_editor.tsx

```diff
@@ -169,8 +169,10 @@ export const NavigationEmbeddableLinkEditor = ({
                 (linkDestination ? defaultLinkLabel : '') ||
                 NavEmbeddableStrings.editor.linkEditor.getLinkTextPlaceholder()
               }
-              value={linkDestination ? currentLinkLabel : ''}
-              onChange={(e) => setCurrentLinkLabel(e.target.value)}
+              value={currentLinkLabel}
```

Comment:
This fixes a bug where a custom title couldn't actually be set when a destination wasn't provided:

**Before:**

https://github.com/elastic/kibana/assets/8698078/b4f15c46-06b7-4194-8a91-fa63e64b0759

**After:**

https://github.com/elastic/kibana/assets/8698078/fb06fda0-203e-4140-819a-851de35889b0